### PR TITLE
RED-226 form instructions overhaul

### PR DESCRIPTION
**Pull request by doombot-exe**
(Almost) All requested changes in RED-226 + overrides to date-picker warnings + hiding settings irrelevant for exams

Missing:
* ENUM for restricting grades view on exam submission
* "cancel" button for reverting changes on the topic settings page

### CODE REVIEW

File: src/Courses/TopicSettings/RendererPreview.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const RendererPreview: React.FC<RendererPreviewProps> = ({defaultPath}) =
     useEffect(()=>{
         if (!_.isNil(defaultPath)) {
             setPreviewSettings(settings => ({...settings, path: defaultPath}));
+            forceUpdate();
```
Comment (red-tt-bear) [inline src/Courses/TopicSettings/RendererPreview.tsx:26]: What does this do?

**Review by red-tt-bear — APPROVED** [review]

File: src/Courses/TopicSettings/RendererPreview.tsx (hunk 1 above)
Comment (PanopticaRising) [reply src/Courses/TopicSettings/RendererPreview.tsx:26]: This forces an update.

Comment (PanopticaRising) [reply src/Courses/TopicSettings/RendererPreview.tsx:26]: Specifically, it generates a new dummy problem object to force the iframe to rerender. A hack to not have to genericize the ProblemIframe component.

**Review by PanopticaRising — APPROVED** [review]

File: src/Courses/TopicSettings/RendererPreview.tsx (hunk 1 above)
Comment (doombot-exe) [reply src/Courses/TopicSettings/RendererPreview.tsx:26]: causes the preview pane to refresh with the new problem path.

**Pull request merged**